src/pyBiodatafuse/graph/rdf.py

```diff
@@ -0,0 +1,1156 @@
+)
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def replace_na_none(item):
+    """
+    Recursively replaces 'na', 'nan', 'none' strings, Python None, and NaN values with None.
+
+    :param item: The value to process (string, None, NaN, list, dict, or np.ndarray).
+    :return: The processed item with 'na', 'nan', 'none', None, and NaN replaced by None.
+    """
+    # Replace 'na' strings
+    if isinstance(item, str) and item.lower() in ["na", "nan", "none"]:
+        return None
+    # Replace Python None (and handle NaNs)
+    elif item is None or (isinstance(item, float) and pd.isna(item)):
+        return None
+    # Recursively handle lists
+    elif isinstance(item, list):
+        return [replace_na_none(sub_item) for sub_item in item]
+    # Recursively handle dictionaries
+    elif isinstance(item, dict):
+        return {key: replace_na_none(value) for key, value in item.items()}
+    # Recursively handle NumPy arrays
+    elif isinstance(item, np.ndarray):
+        return np.array([replace_na_none(sub_item) for sub_item in item], dtype=object)
+    # Return the item as-is if no replacement needed
+    else:
+        return item
+
+
+def add_gene_protein_nodes(g: Graph, row) -> URIRef:
+    """Create and add a gene node to the RDF graph.
+
+    :param g: RDF graph to which the gene node will be added.
+    :param row: DataFrame row containing gene information.
+    :return: URIRef for the created gene node.
+    """
+    target = row["target"]
+    if target:
+        gene_node = URIRef(f"http://identifiers.org/ensembl/{target}")
+        protein_node = URIRef(f"http://identifiers.org/ensembl/{target}_protein")
+        # TODO replace by UniProt ID when available
+        g.add((gene_node, RDF.type, URIRef(NODE_TYPES["gene_node"])))
+        g.add((gene_node, RDFS.label, Literal(row["identifier"], datatype=XSD.string)))
+        g.add((protein_node, URIRef(PREDICATES["has_gene_template"]), gene_node))
+        g.add((protein_node, RDF.type, URIRef(NODE_TYPES['protein_node'])))
+        return [gene_node, protein_node]
+    else:
+        return None
+
+
+def add_disease_node(g: Graph, disease_data: dict) -> URIRef:
+    """Create and add a disease node to the RDF graph.
+
+    :param g: RDF graph to which the disease node will be added.
+    :param disease_data: Dictionary containing disease information.
+    :return: URIRef for the created disease node.
+    """
+    # UMLS IRIs not in Bioregistry
+    disease_curie = disease_data.get("disease_umlscui")
+    if disease_curie is None:
+        disease_curie = disease_data.get("UMLS")
+    disease_iri = f"https://www.ncbi.nlm.nih.gov/medgen/{disease_curie}"
+    disease_node = URIRef(disease_iri)
+    g.add((disease_node, RDF.type, URIRef(NODE_TYPES["disease_node"])))
+    g.add(
+        (disease_node, RDFS.label, Literal(disease_data.get("disease_name"), datatype=XSD.string))
+    )
+    for identifier_type in [
```

Comment:
this could be called from constant file, to make sure if they change latter, it would be reflected here too